### ISSUE TITLE
NDN-lite main event loop vs. dedicated thread

### ISSUE THREAD

**Issue by cgundogan** [named-data-iot/ndn-lite-over-riot#1]
In the current implementation, `ndn-lite` relies on an event loop in the main application (e.g., [here](https://github.com/named-data-iot/ndn-lite-over-riot/blob/master/examples/ndn-consumer/main.c#L77)). A more suitable way in RIOT is to create a dedicated `ndn-lite` thread that runs the process manager and frees the main thread to be used by the user application.

Was there a specific design decision for not using a dedicated thread?

@yan-foto initially encountered this issue.

**Comment by cgundogan** [comment]
Any ideas on how to proceed here, @RoySCU ?

**Comment by tianyuan129** [comment]
I agree this is a more suitable way for RIOT application and also openthread and ndn-riot were packaged in this way (if I remember correctly).

We did not using a dedicated thread for trying to keep the same style with other environment's implementation (e.g., [POSIX](https://github.com/named-data-iot/ndn-iot-package-over-posix)). But I indeed agree switching to a dedicated thread is more suitable here.
